File: ProjectPos.BackOffice.FrontEnd/src/proxy/interfaces/product-inventory-dto.ts
import { FullAuditedEntityDto } from "../entity-dtos/full-audited-entity-dto";
import { Category } from "../enums/category";
import { Grade } from "../enums/grade";
import { Status } from "../enums/status";
import { Unit } from "../enums/unit";
import { ProductDto } from "./product-dto";
import { ProductPriceDto } from "./product-price-dto";

export interface ProductInventoryDto extends FullAuditedEntityDto<number> {
    productId?: number;
    barCode?: string;
    description?: string
    subCategoryName?: string;
    pLUCode?: string;
    flag?: number;
    isWeighted?: boolean;
    name?: string;
    grade?: Grade;
    idealQuantity?: number;
    quantityOnHand?: number;
    quantityOnShelf?: number;
    stockCount?: number;
    quantity?: number;
    cost?: number;
    markUp?: number;
    unit?: Unit;
    status?: Status;
    category?: Category;
    product?: ProductDto;
    isTaxable?:boolean
    subCategoryId?: number;
    
    productPrice?: ProductPriceDto;
}

File: ProjectPos.BackEnd/ProjectPos.Services/DTOs/ProductInventoryDto.cs
using ProjectPos.Services.EntityDtos;
using ProjectPos.Data.Shared.Enums;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ProjectPos.Services.DTOs
{
    public class ProductInventoryDto : FullAuditedEntityDto<int>
    {
        public string? BarCode { get; set; }
        public string? SubCategoryName { get; set; }
        public string? Name { get; set; }
        public int Flag { get; set; } = 20;
        public string? PLUCode { get; set; }
        public bool? IsWeighted { get; set; } = false;
        public bool? IsTaxable { get; set; }
        public string? Description { get; set; }
        public Category? Category { get; set; }
        public int? SubCategoryId { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Cost { get; set; }
        public Grade? Grade { get; set; }
        public decimal? IdealQuantity { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? QuantityOnHand { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? QuantityOnShelf { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? StockCount { get; set; }
        [Column(TypeName = "decimal(12, 2)")]
        public decimal? Quantity { get; set; }
        public Unit? Unit { get; set; }
        public Status? Status { get; set; }
        public decimal? MarkUp { get; set; }

        public ProductPriceDto? ProductPrice { get; set; }
    }
}
